Red_Black_Tree/RedBlackTree.java

```diff
@@ -0,0 +1,331 @@
+                       root=root.parent;
+                       rotate(root, true,"right");
+                   }
+               }
+               else
+               {
+                   root.color=BLACK;
+                   sibling.color=BLACK;
+                   if(root.parent.parent!=null)
+                   {
+                       root.parent.color=RED;
+                   }
+               }
+           } 
+        }
+        return root;
+    }
+     private Node findSiblingNode(Node root) {
+        Node parent = root.parent;
+        if(parent==null)
+        {
+            return null;
+        }
+        if(isLeftChild(root)) 
+        {
+            return parent.right.isNull ? null : parent.right;
+        } 
+        else 
+        {
+            return parent.left.isNull ? null : parent.left;
+        }
+    }
+    
+    private void printInOrder(Node root)
+    {
+        if (root == null || root.isNull) 
+            return; 
+        
+        printInOrder(root.left); 
+        System.out.print(root.data +(root.color == BLACK ? "B" : "R")+" "); 
+        printInOrder(root.right); 
+    }
+
+    private void displayRedBlackTree(Node root, int space) {
+        
+        if(root == null || root.isNull) {
+            return;
+        }
+        displayRedBlackTree(root.right, space + 20);
+        for(int i=0; i < space; i++) {
+            System.out.print(" ");
+        }
+        System.out.println(root.data + " " + (root.color == BLACK ? "B" : "R"));
+        
+        
+        displayRedBlackTree(root.left, space + 20);
+    }
+    public static void main(String args[])
+    {   
+        Node root = null;
+        RedBlackTree redBlackTree = new RedBlackTree();
+
+        root = redBlackTree.insert(root, 10);
+        root = redBlackTree.insert(root, 25);
+        root = redBlackTree.insert(root, 5);
+        root = redBlackTree.insert(root, 28);
+        root = redBlackTree.insert(root, 76);
+        root = redBlackTree.insert(root, 55);
+        root = redBlackTree.insert(root, 50);
+        root = redBlackTree.insert(root, 15);
+        root = redBlackTree.insert(root, 18);
+        
+        redBlackTree.displayRedBlackTree(root,0);
+        /* 10 is the root node
+        OUTPUT:
+                                                                            76 R
+                                                55 B
+                                                                    50 R
+                            28 R
+                                                                    25 R
+                                                18 B
+                                                                    15 R
+        10 B
+                            5 B
+        */
+        System.out.println("Inorder traversal:");
+        redBlackTree.printInOrder(root);
+        /* Inorder traversal:
+           5B 10B 15R 18B 25R 28R 50R 55B 76R */
+        System.out.println();
+    }
+}
```

Comment:
Add after line 331 in order:
- Single blank line
- Sample input-output
- Single blank line